### PR TITLE
Type tuple error messages

### PR DESCRIPTION
**Pull request by daviditen**
Add user errors for accessing a tuple of types out of bounds.  This was previously an
internal error.

Add a user error for passing too many arguments to a type-tuple indexing expression.
This was previously allowed to compile and run with the extra arguments ignored.

Add new tests for each of these errors.

### CODE REVIEW

**Comment by daviditen** [comment]
Passed a full Linux64 paratest.

File: compiler/resolution/preFold.cpp [hunk 1]
```diff
@@ -1123,14 +1123,27 @@ static Expr* preFoldNamed(CallExpr* call) {
...
+
+      if (!at) {
         USR_FATAL(call, "illegal type index expression");
       }
 
+      if (index <= 0 || index > at->fields.length-1) {
+        char msg[60];
+        snprintf(msg, 60,
+                 "type index expression '%" PRId64 "' out of bounds", index);
+        USR_FATAL(call, msg);
```
Comment (mppf) [inline compiler/resolution/preFold.cpp]: Shouldn't this just be USR_FATAL(call, "type index expression '%i out of bounds", (int) index) ?

Comment (daviditen) [reply compiler/resolution/preFold.cpp]: Yes, that sounds better to me.  I'll update it.

**Comment by daviditen** [comment]
Thanks @mppf!  I just pushed your suggested improvement.

**Pull request merged**